docs/resources/etc_fstab.md.erb

```diff
@@ -0,0 +1,126 @@
+* `device_name` is the name associated with the device.
+* `mount_point` is the default directory the device will be mounted if one is not specified in the mount command.
+* `file_system_type` is the type of file system of the device or partition.
+* `mount_options` is the options for the device or partition.
+* `dump_options` is a number used by dump to decide if a file system should be backed up.
+* `file_system_options` is a number that specifies the order the file system should be checked.
+
+## Property Examples and Return Types
+
+### device_name(String)
+
+`device_name` returns an array of strings that matches the where condition of the filter table.
+
+    describe etc_fstab.where { mount_point == '/mnt/sr0' } do
+      its ( 'device_name' ) { should eq ['/dev/sr0'] }
+    end
+
+### mount_point(String)
+
+`mount_point` returns an array of strings that matches the where condition of the filter table.
+
+    describe etc_fstab.where { device_name == '/dev/sr0' } do
+      its('mount_point') { should eq ['/mnt/sr0'] }
+    end
+
+### file_system_type(String)
+
+`file_system_type` returns an array of strings that matches the where condition of the filter table.
+
+    describe etc_fstab.where { device_name == '/dev/sr0' } do
+      its('file_system_type') { should eq ['iso9660'] }
+    end
+
+### mount_options([String])
+
+`mount_options` returns a two dimensional array of strings that matches the where condition of the filter table.
+
+    describe etc_fstab.where { mount_point == '/' } do
+      its('mount_options') { should eq [['defaults', 'x-systemd.device-timeout=0']] }
+    end
+
+### dump_options(int)
+
+`dump_options` returns an array of strings that matches the where condition of the filter table.
+
+    describe etc_fstab.where { device_name == '/dev/sr0' } do
+      its('dump_options') { should eq [0] }
+    end
+
+### file_system_type(int)
+
+`file_system_options` returns an array of strings that matches the where condition of the filter table.
+
+    describe etc_fstab.where { device_name == '/dev/sr0' } do
+      its('file_system_options') { should eq [0] }
+    end
+
+## Examples
+
+The following examples show how to use this InSpec resource.
+
+### Check all partitions that have type of 'nfs'.
+
+    nfs_systems = etc_fstab.nfs_file_systems
+    nfs_systems.each do |partition|
+      describe partition do
+        its ( 'mount_options' ) { should include 'nosuid' }
+      end
+    end
+
+### Check all partitions that are removable file systems.
+
+    removable_media = etc_fstab.removable_media_file_systems
+    removable_media.each do |media|
+      describe media do
+        its ( 'mount_options' ) { should include 'nosuid' }
+      end
+    end
+
+### Check the partition mounted at /home contains 'nosuid' in its mount_options.
+
+    describe etc_fstab do
+      its ('home_mount_options') { should include 'nosuid' }
+    end
+
+### Check if a partition is mounted at a point.
+
+    describe etc_fstab.mounted?('/home') do
```

Comment:
This will output better in the formatter if written with a .where:

```ruby
describe etc_fstab.where { mount_point == '/home' } do
  it { should be_mounted }
end
```